miner/src/PoolMiner.java

```diff
@@ -127,6 +130,26 @@ private void subscribe() throws Exception
 
     String host = config.get("pool_host");
     int port = config.getIntWithDefault("pool_port", 23380);
+
+    String[] hosts = host.split(";|,");
+
+    host = hosts[nextHostIndex];
+    nextHostIndex = (nextHostIndex + 1) % hosts.length;
```

Comment:
I was thinking that since the connection and subscribe is cheap for everyone, to connect and sunscribe to each pool and have the mining threads pick up the work from the first pool with valid work.

That way we also return to first pool after fail over quickly.